Dialog/CreditsEditDialog.cpp

```diff
@@ -0,0 +1,277 @@
+
+    for (int k=0;k<13;k++) {
+        model[k]=new QStandardItemModel();
+
+        memcpy(data_to_save[k],&ROMUtils::CurrentFile[0x789FCC+k*1280],1280); //1280 = 0x500 size of one credit screen
+
+        for (int i=0;i<20;i++) {
+            for (int j=0;j<32;j=j+1) {
+                QStandardItem *newItem = new QStandardItem();
+
+                //The two last column never appeared in game
+                if (j>=30) {
+                    newItem->setEditable(false);
+                    newItem->setData(QColor(Qt::gray), Qt::BackgroundRole);
+                } else {
+                    short mapId=data_to_save[k][j+i*32];
+                    newItem->setText(QString(this->CreditTileMap[mapId]));
+
+                    //One-tile high -> blue
+                    if (mapId >= 0x4340 && mapId <= 0x435B) {
+                        newItem->setData(QColor("#8080ff"), Qt::BackgroundRole);
+                    //Two-tile high, upper half -> red
+                    } else if ((mapId >= 0x0360 && mapId <=0x0379) || (mapId >= 0x03A0 && mapId <= 0x03B9)) {
+                        newItem->setData(QColor("#ff8080"), Qt::BackgroundRole);
+                    //Two-tile high, lower half -> green
+                    } else if ((mapId >= 0x0380 && mapId <=0x0399) || (mapId >= 0x03C0 && mapId <= 0x03D9)) {
+                        newItem->setData(QColor("#80ff80"), Qt::BackgroundRole);
+                    }
+
+                }
+                model[k]->setItem(i, j, newItem);
+            }
+        }
+    }
+
+    for (int i=0 ;i<13;i++) {
+        tabs[i]=new QWidget;
+        tables_view[i]=new CreditEditor_TableView(tabs[i]);
+
+        tables_view[i]->setModel(model[i]);
+
+        //Needed in ordrer to have the smallest cells
+        for (int j=0;j<32;j++) {
+            tables_view[i]->setColumnWidth(j,1);
+        }
+
+        for (int j=0;j<20;j++) {
+            tables_view[i]->setRowHeight(j,1);
+        }
+
+        tables_view[i]->setObjectName(QString::number(i+1));
+
+        //With this grid Layout the CustomQTableView takes all available size in the tab
+        QGridLayout* layout = new QGridLayout;
+        layout->addWidget(tables_view[i], 0,0);
+        tabs[i]->setLayout(layout);
+        ui->tabWidget->addTab(tabs[i],QString::number(i+1));
+    }
+
+}
+
+/// <summary>
+/// Deconstructor of CreditsEditDialog class.
+/// </summary>
+CreditsEditDialog::~CreditsEditDialog()
+{
+    delete ui;
+}
+
+/// <summary>
+/// Perform static initialization of constant data structures for the dialog.
+/// </summary>
+void CreditsEditDialog::StaticInitialization()
+{
+    // Initialize the selections for the Door type
+    for (unsigned int i = 0; i < sizeof(CreditTileMapData) / sizeof(CreditTileMapData[0]); i+=2)
+    {
+        CreditTileMap[CreditTileMapData[i]] = static_cast<char>(CreditTileMapData[i + 1]);
+    }
+
+    //One-tile high
+    CreditTileReverseMap.insert(std::make_pair(std::make_pair("A","#8080ff"),0x4340));
```

Comment:
This section is bad. Let's construct `CreditTileReverseMap` in the above for loop, `#define` the color strings, and make use of the [] operator overloading instead of .insert()